src/webui/api/synccontroller.cpp

```diff
@@ -384,6 +406,41 @@ namespace
 
         return QJsonObject::fromVariantMap(syncData);
     }
+
+    void addAnnounceStats(QVariantMap &serializedTorrent, const BitTorrent::Torrent *torrent)
+    {
+        bool hasTrackerWarning = false;
+        bool hasTrackerError = false;
+        bool hasOtherAnnounceError = false;
+        for (const BitTorrent::TrackerEntryStatus &status : asConst(torrent->trackers()))
+        {
+            switch (status.state)
+            {
+            case BitTorrent::TrackerEndpointState::Working:
+                if (!hasTrackerWarning && hasWarningMessage(status))
+                    hasTrackerWarning = true;
+                break;
+            case BitTorrent::TrackerEndpointState::TrackerError:
+                if (!hasTrackerError)
```

Comment:
is this really needed? assigning it every time is more costly or what?

Reply:
> is this really needed? assigning it every time is more costly or what?

"assigning it every time" just seems counterintuitive to me. I did not measure their performance.

Reply:
I think this additional checking can be omitted in the cases where it doesn't guard against doing some other (unnecessary) job.